system-test/automation_utils.sh

```diff
@@ -74,6 +74,20 @@ function wait_for_bootstrap_validator_stake_drop {
   ssh "${sshOptions[@]}" "${validatorIpList[0]}" "RUST_LOG=info \$HOME/.cargo/bin/solana wait-for-max-stake $max_stake --url http://127.0.0.1:8899"
 }
 
+function wait_for_equal_stake {
+  source "${REPO_ROOT}"/net/common.sh
+  loadConfigFile
+
+  max_stake=$((100 / ${#validatorIpList[@]} + 1))
+  execution_step "Waiting for stake on all validators to fall below ${max_stake}%"
+
+  # shellcheck disable=SC2154
+  for ip in "${validatorIpList[@]}"; do
```

Comment:
Is the looping necessary?  `solana wait-for-max-stake` waits for stake visibility in gossip, so you should be fine to just calculate `max_stake` here, then call `wait_for_bootstrap_validator_stake_drop` (which has a very antiquated name now :sweat_smile:)

Reply:
ah you're totally right, I got fooled by the name :)

Reply:
It won't take you long to identify our pattern of land mines :wink: